File: frontend/src/types/supplyChain.ts
export type ResourceBalanceStatus = "SURPLUS" | "NEUTRAL" | "NEEDS_RESOURCE"

export type TransferStatus = "PROPOSED" | "APPROVED" | "REJECTED" | "IN_TRANSIT" | "DELIVERED"

export interface FacilityResourceBalance {
  facility_id: string
  facility_name: string
  state_name: string
  district_name: string
  facility_type: string
  balance_status: ResourceBalanceStatus
  resilience_score: number
  surplus_items: string[]
  deficit_items: string[]
}

export interface TransferRecommendation {
  recommendation_id: string
  source_facility_id: string
  source_facility_name: string
  target_facility_id: string
  target_facility_name: string
  medicine_id: string
  medicine_name: string
  batch_number: string
  quantity: number
  unit: string
  distance_km: number
  estimated_transit_hours: number
  match_confidence: number
  urgency_level: "CRITICAL" | "HIGH" | "MODERATE"
  reasoning: string
  status: TransferStatus
  created_at: string
  decided_at?: string | null
  decided_by?: string | null
}

export interface Shipment {
  shipment_id: string
  origin_name: string
  destination_name: string
  medicine_name: string
  quantity: number
  unit: string
  status: "DISPATCHED" | "IN_TRANSIT" | "DELAYED" | "DELIVERED"
  eta: string
  delay_days: number
  corridor_status: "CLEAR" | "CONGESTED" | "WEATHER_DELAY"
  carrier_name: string
}

export interface SupplyChainSummary {
  total_shipments_in_transit: number
  surplus_nodes_count: number
  deficit_nodes_count: number
  pending_recommendations_count: number
  recommendations: TransferRecommendation[]
  shipments: Shipment[]
  facility_balances: FacilityResourceBalance[]
}

File: backend/app/schemas/supply_chain.py
from enum import Enum
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field

class ResourceBalanceStatus(str, Enum):
    SURPLUS = "SURPLUS"
    NEUTRAL = "NEUTRAL"
    NEEDS_RESOURCE = "NEEDS_RESOURCE"

class TransferStatus(str, Enum):
    PROPOSED = "PROPOSED"
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"
    IN_TRANSIT = "IN_TRANSIT"
    DELIVERED = "DELIVERED"

class FacilityResourceBalance(BaseModel):
    facility_id: str
    facility_name: str
    state_name: str
    district_name: str
    facility_type: str
    balance_status: ResourceBalanceStatus
    resilience_score: float
    surplus_items: List[str] = []
    deficit_items: List[str] = []

class TransferRecommendation(BaseModel):
    recommendation_id: str
    source_facility_id: str
    source_facility_name: str
    target_facility_id: str
    target_facility_name: str
    medicine_id: str
    medicine_name: str
    batch_number: str
    quantity: int
    unit: str
    distance_km: float
    estimated_transit_hours: float
    match_confidence: float = Field(..., ge=0.0, le=1.0)
    urgency_level: str  # "CRITICAL", "HIGH", "MODERATE"
    reasoning: str
    status: TransferStatus = TransferStatus.PROPOSED
    created_at: str
    decided_at: Optional[str] = None
    decided_by: Optional[str] = None

class Shipment(BaseModel):
    shipment_id: str
    origin_name: str
    destination_name: str
    medicine_name: str
    quantity: int
    unit: str
    status: str  # "DISPATCHED", "IN_TRANSIT", "DELAYED", "DELIVERED"
    eta: str
    delay_days: int = 0
    corridor_status: str  # "CLEAR", "CONGESTED", "WEATHER_DELAY"
    carrier_name: str

class SupplyChainSummary(BaseModel):
    total_shipments_in_transit: int
    surplus_nodes_count: int
    deficit_nodes_count: int
    pending_recommendations_count: int
    recommendations: List[TransferRecommendation]
    shipments: List[Shipment]
    facility_balances: List[FacilityResourceBalance]
